source/common/network/io_socket_handle_impl.cc

```diff
@@ -114,6 +114,16 @@ Api::IoCallUint64Result IoSocketHandleImpl::read(Buffer::Instance& buffer,
   if (max_length == 0) {
     return Api::ioCallUint64ResultNoError();
   }
+  if constexpr (Event::PlatformDefaultTriggerType == Event::FileTriggerType::EmulatedEdge) {
+    if (buffer_->length() > 0) {
+      auto move_length = std::min(buffer_->length(), max_length);
+      buffer.move(*buffer_);
+      if (file_event_) {
```

Comment:
I am not sure if this is needed. The caller should be reading until `EAGAIN` is returned